lib/wrappers/output.ts

```diff
@@ -0,0 +1,61 @@
+import { PathLike } from "fs";
+import TransformNodeWrapper from "./transform-node";
+
+const fsExtra = require('fs-extra');
+const fs = require('fs');
+const logger = require('heimdalljs-logger')('broccoli:outputWrapper');
+const path = require('path');
+
+const WRITEOPERATION = new Set([
+  'write',
+  'writeSync',
+  'writeFile',
+  'writeFileSync',
+  'writev',
+  'writevSync',
+  'appendFileSync',
+  'appendFile',
+  'rmdir',
+  'rmdirSync',
+  'mkdir',
+  'mkdirSync',
+  'createWriteStream',
+  'copyFile',
+  'copyFileSync'
+]);
+
+export default class OutputWrapper {
```

Comment:
@SparshithNR  Would something like this work?

```js
const fs = require('fs');

const WHITE_LIST = new Set([
  'writeFileSync'
]);

export default function OutputWrapper(node) {
  return new Proxy(fs, {
    get(target, propertyName) {
      if (!WHITE_LIST.has(propertyName) || !target.hasOwnProperty(propertyName)) {
        throw new Error(`${propertyName} method is not allowed. Supported methods are: ${WHITE_LIST.join(', ')}`);
      }

      return () => {
        // check if the first arg is path like and warn?

        target[propertyName](node.outputPath, ...arguments);
      };
    }
  };
}
```

Reply:
This will work. But I feel that, it will become hard to extend the features of OutputWrapper like, we want to add more than just a proxy to `fs` in output wrapper. For example, if we want to add one more proxy like `path`, we will not able to add it directly or if we want to add a revision or rebuild functions.
If we are not expecting this. Above snippet will work. :)